mlflow/xgboost.py

```diff
@@ -184,3 +193,102 @@ def __init__(self, xgb_model):
     def predict(self, dataframe):
         import xgboost as xgb
         return self.xgb_model.predict(xgb.DMatrix(dataframe))
+
+
+@experimental
+def autolog(importance_types=['weight']):  # pylint: disable=W0102
+    """
+    Enables autologging from XGBoost to MLflow. Logs the following.
```

Comment:
Would be good to mention that this `autolog` API only supports the XGBoost Booster API